src/Controls/src/Core/Shell/Shell.cs

```diff
@@ -1661,6 +1661,14 @@ static void OnCurrentItemChanged(BindableObject bindable, object oldValue, objec
 			shell.ShellController.AppearanceChanged(shell, false);
 			shell.ShellController.UpdateCurrentState(ShellNavigationSource.ShellItemChanged);
 
+			// If the new ShellItem is a FlyoutItem or a ShellGroupItem, we need to update the QueryAttributes
+			var existing = (ShellRouteParameters)shell.CurrentContent?.GetValue(ShellContent.QueryAttributesProperty);
+            if(oldValue is not null && newValue is not null && (shell.CurrentItem is FlyoutItem || shell.CurrentItem is ShellGroupItem) && existing is null)
```

Comment:
This is a condition with a lot of checks. Not required but, we could improve the readability with small changes:

```
bool isRelevantShellItem = shell.CurrentItem is FlyoutItem || shell.CurrentItem is ShellGroupItem;
bool isNewValueDifferent = oldValue is not null && newValue is not null && existing is null;

if (isNewValueDifferent && isRelevantShellItem)
```

Reply:
Hi @jsuarezruiz  , I have modified the condition based on your suggestion. Kindly review and let me know if you have any concern.